### PR TITLE
Feature/simulate occupancy

### LINKED ISSUE #14: Stats Module - Simulated Fullness API
Create an endpoint that returns a simulated "real-time" bus fullness percentage for a given bus ID.

### PR DESCRIPTION
**Pull request by LUCiv175**
This pull request adds a bus occupancy simulation feature to the backend, integrating real-time and expected occupancy data into trip responses. It introduces a new `Bus` model and seeding logic, enhances the `/stops/:id` endpoint to provide enriched trip details including bus capacity/type and occupancy estimates, and implements a configurable occupancy simulation algorithm. Comprehensive tests have also been added to ensure correct behavior.

**Bus Model and Data Initialization**
* Added a new `Bus` model (`backend/src/models/Bus.js`) to store bus capacity and type information, and implemented a seeding script to populate the database with bus details (`backend/src/seedBus.js`). The server now seeds the bus collection on startup (`backend/src/server.js`). [[1]](diffhunk://#diff-240b9724268bfc0223e772b8d9b2e6f5b357cdea6559926e656825002800ae26R1-R13) [[2]](diffhunk://#diff-eab099d6299292b4ef0770c62744b4bd38b0a23f66cb3224850e2afd2567db51R1-R338) [[3]](diffhunk://#diff-36e2c2dd1e67a7419cef780285f514e743e48ac994a01526288acd31707e09aeR4-R10)

**Trip Endpoint Enhancements**
* The `/stops/:id` endpoint now enriches each trip with bus capacity/type and both real-time and expected occupancy estimates, using data from the new `Bus` model and the occupancy simulation utility. Trip responses include new fields such as `busCapacity`, `busType`, `occupancyRealTime`, and `occupancyExpected`.

**Occupancy Simulation Logic**
* Implemented a configurable occupancy simulation algorithm (`backend/src/utils/simulateOccupancy.js`) that estimates bus load based on route configuration, time of day, day of week, and trip progress. Route profiles and peak locations are defined in a new config file (`backend/src/utils/routeConfig.js`). [[1]](diffhunk://#diff-54929217940975904b3f042eeea171064941f0e1ec18724d6af32d4b8aa0a4f6R1-R107) [[2]](diffhunk://#diff-98e27faf02582946967db8734834fadd47363a85fd3936703fcce1d3d1d8b6d9R1-R94)

**Testing**
* Added and updated tests to validate the `/stops/:id` endpoint's new occupancy data and the correctness of the occupancy simulation logic, including edge cases and color-level assignments. [[1]](diffhunk://#diff-c28957170c78be2bbc64a470c0d647f5a56f4ee6fcf77925a90b3e452f86502eL63-R63) [[2]](diffhunk://#diff-c28957170c78be2bbc64a470c0d647f5a56f4ee6fcf77925a90b3e452f86502eL108-R125) [[3]](diffhunk://#diff-543affb1c24af08d5f1f4c948d0f720bdb4d3c14e083deedaaf1f919a03d20b6R1-R122)

### CODE REVIEW

File: backend/tests/simulateOccupancy.test.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    expect(result.passengers).toBeGreaterThanOrEqual(0);
+    expect(result.passengers).toBeLessThanOrEqual(100);
+  });
+
+  test("Dovrebbe gestire correttamente la prima fermata (stopIndex 0)", () => {
+    getRouteConfig.mockReturnValue({ profile: "standard" });
+    const result = simulateOccupancy(new Date(), "tripStart", 100, 0, 20, "r1");
+
+    expect(result.percentage).toBe(10);
+  });
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/tests/simulateOccupancy.test.js]: outdated

File: backend/src/utils/simulateOccupancy.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    timeFactor = Math.max(morning, lunch, evening, 0.05);
+  }
+
+  if (dayOfWeek === 0) timeFactor *= 1.1; // Sunday
+  if (dayOfWeek === 1) timeFactor *= 0.8; // Monday
+  if (dayOfWeek === 2) timeFactor *= 0.9; // Tuesday
+  if (dayOfWeek === 3) timeFactor *= 1.0; // Wednesday
+  if (dayOfWeek === 4) timeFactor *= 0.9; // Thursday
+  if (dayOfWeek === 5) timeFactor *= 0.8; // Friday
+  if (dayOfWeek === 6) timeFactor *= 0.9; // Saturday
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/src/utils/simulateOccupancy.js]: done

File: backend/src/app.js [hunk 3]
```diff
@@ -48,51 +50,131 @@ app.get("/stops", (req, res) => {
...
+        : [];
+      const totalStops = stopTimes.length;
+
+      const lastStopId = Number(element.stopLast);
+      let currentBusIndex = -1;
+
+      if (lastStopId === 0) {
+        currentBusIndex = 0;
+      } else {
+        currentBusIndex = stopTimes.findIndex((s) => s.stopId === lastStopId);
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/src/app.js]: that's incorrect

File: backend/src/utils/routeConfig.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    // Line 14 from P.Dante to Belvedere
+    profile: "draining",
+    peakLocation: 0.2,
+  },
+
+  536: {
+    // Line 17 to Dogana
+    profile: "accumulator",
+    peakLocation: 0.9,
+  },
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/src/utils/routeConfig.js]: done

File: backend/tests/app.test.js [hunk 5]
```diff
@@ -1,185 +1,152 @@
...
+      }
     ];
 
-    const mockRoute = {
-      routeId: "R1",
-      routeShortName: "10",
-      routeLongName: "Centro - Sobborgo",
-      routeColor: "C52720",
-    };
+    it("It would correctly process data and calculate occupancy", async () => {
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/tests/app.test.js]: done

File: backend/src/utils/simulateOccupancy.js [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+  let hash = 0;
+  for (let i = 0; i < seedString.length; i++) {
+    hash = ((hash << 5) - hash) + seedString.charCodeAt(i);
+    hash |= 0;
+  }
+  const randomVariance = ((Math.abs(hash) % 100) / 500) - 0.1;
+
+  let finalLoadFactor = rawLoad + randomVariance;
+
+  // Clamp e Output
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/src/utils/simulateOccupancy.js]: done

File: backend/src/seedBus.js [hunk 7]
```diff
@@ -0,0 +1,338 @@
...
+  try {
+    await Bus.deleteMany({});
+    console.log('🧹 Collezione "buses" pulita.');
+
+    const result = await Bus.insertMany(busData);
+    console.log(
+      `🚀 Inseriti ${result.length} bus nel database automaticamente!`
+    );
+  } catch (error) {
+    console.error("❌ Errore durante il seeding automatico:", error);
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/src/seedBus.js]: done

File: backend/src/utils/simulateOccupancy.js [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+  } else {
+    const morning = getPeakFactor(decimalTime, 7.5, 1.5);
+    const lunch = getPeakFactor(decimalTime, 13.5, 1.5) * 0.8;
+    const evening = getPeakFactor(decimalTime, 17.5, 2.0) * 0.9;
+    timeFactor = Math.max(morning, lunch, evening, 0.05);
+  }
+
+  let dayMultiplier = 1.0;
+  switch (dayOfWeek) {
+    case 0: dayMultiplier = 1.1; break; // Sunday
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/src/utils/simulateOccupancy.js]: false

File: backend/tests/app.test.js [hunk 9]
```diff
@@ -1,185 +1,152 @@
...
-      status: 503,
-      statusText: "Service Down",
+      expect(res.statusCode).toBe(200);
+      const trip = res.body[0];
+      expect(trip.busCapacity).toBe(100);
+      expect(trip.busType).toBe("standard");
     });
 
-    const res = await request(app).get("/stops/999");
+    it("It would return 502 if the external API fails", async () => {
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/tests/app.test.js]: done

File: backend/src/seedBus.js [hunk 10]
```diff
@@ -0,0 +1,338 @@
...
+];
+
+export async function seedDatabase() {
+  try {
+    await Bus.deleteMany({});
+    console.log('🧹 Collezione "buses" pulita.');
+
+    const result = await Bus.insertMany(busData);
+    console.log(
+      `🚀 Inseriti ${result.length} bus nel database automaticamente!`
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/src/seedBus.js]: done

File: backend/tests/app.test.js [hunk 11]
```diff
@@ -1,185 +1,152 @@
...
-
-    const sortByStopId = (arr) =>
-      arr.slice().sort((a, b) => {
-        if (a.stopId < b.stopId) return -1;
-        if (a.stopId > b.stopId) return 1;
-        return 0;
-      });
-    expect(sortByStopId(response.body)).toEqual(sortByStopId(expectedStops));
+  describe("GET /", () => {
+    it("It would respond with status 200 and a welcome message", async () => {
```
[the remark this answers is not in the document]
Comment (LUCiv175) [reply backend/tests/app.test.js]: done

**Review by Ceralex — CHANGES_REQUESTED** [review]
I think busId should be treated as an integer instead of a String, to reflect the rest of the api specification

**Review by Ceralex — APPROVED** [review]
